File: Python/개인정보 수집 유효기간.py
def init_date(today):
	global year
	global month
	global day
	year = int(today[0:4])
	month = int(today[5:7])
	day = int(today[8:10])

def check_date(date, num):
	_year = int(date[0:4])
	_month = int(date[5:7])
	_day = int(date[8:10])
	_month += num
	if _month > 12:
		_year += 1
		_month -= 12
	print(year,month,day)
	print(_year,_month,_day)
	if year < _year:
		return False
	if year <= _year and month < _month:
		return False
	if year <= _year and month <= _month and day < _day :
		return False
	return True
	
def solution(today, terms, privacies):
	answer = []
	dic = {}
	init_date(today)
	for t in terms:
		dic[t.split()[0]] = int(t.split()[1])
	for date in privacies:
		print(check_date(date.split()[0], dic[date.split()[1]]))
	print(dic)
	return answer



today = "2022.05.19"
terms = ["A 6", "B 12", "C 3"]
privacies=["2021.05.02 A", "2021.07.01 B", "2022.02.19 C", "2022.02.20 C"]

solution(today, terms, privacies)

File: C++/Prgrammers_LV1/개인 정보 수집 유효기간.cpp
#include <string>
#include <vector>
#include <sstream>
#include <iostream>
#include <map>
using namespace std;


vector<string> split(string input, char delimiter) {
    vector<string> answer;
    stringstream ss(input);
    string temp;
 
    while (getline(ss, temp, delimiter)) {
        answer.push_back(temp);
    }
    return answer;
}

int	check_date(vector<string> date, string c_date, int num)
{
	vector<string> _date = split(c_date, '.');
	int year = stoi(date[0]);
	int	month = stoi(date[1]);
	int	day = stoi(date[2]);
	int _year = stoi(_date[0]);
	int	_month = stoi(_date[1]);
	int	_day = stoi(_date[2]);

	_month += num;
	while (_month > 12)
	{
		_year++;
		_month -= 12;
	}
	if (year > _year)
		return 0;
	else if (year == _year)
	{
		if (month > _month)
			return 0;
		else if (month == _month)
		{
			if (day >= _day)
				return 0;
		}
	}
	return 1;
}

vector<int> solution(string today, vector<string> terms, vector<string> privacies) {
    vector<int>				answer;
	vector<string>	date = split(today, '.');
	map<string, int>		dic;
	int idx = 0;
	
	for (auto t : terms)
	{
		vector<string> tmp = split(t, ' ');
		dic[tmp[0]] = stoi(tmp[1]);
	}
	for (auto p : privacies)
	{
		vector<string> tmp = split(p, ' ');
		if (check_date(date, tmp[0], dic[tmp[1]]) == 0)
			answer.push_back(idx+1);
		idx++;
	}
    return answer;
}


#include "printVector.cpp"
int  main()
{
	string today = "2022.05.19"	;
	vector<string> terms = {"A 6", "B 12", "C 3"};
	vector<string> privacies = {"2021.05.02 A", "2021.07.01 B", "2022.02.19 C", "2022.02.20 C"};
	printVector(solution(today, terms, privacies));
}